sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -107,7 +107,7 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
 						"time_range": {
 							Type:         schema.TypeString,
 							Optional:     true,
-							ValidateFunc: validation.StringMatch(regexp.MustCompile(`-?(\d)+[smhd]`), "Time range must be in the format '-?\\d+[smhd]'. Examples: -15m, 1d, etc."),
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^-?(\d)+[smhd]$`), "Time range must be in the format '-?\\d+[smhd]'. Examples: -15m, 1d, etc."),
```

Comment:
we don't want to force user to use -ve values ATM. 

Current logic basically takes -ve/+ve values but when the response comes from api, we are converting the time range to a positive value this will lead to some issues where tf state would not match and tf will keep applying.

Reply:
> we don't want to force user to use -ve values ATM.

How is the change forcing the `-` sign? I fail to see it.

Reply:
I agree that having choice here between +ve/-ve will lead to TF problems (state won't change after apply), but it is solved in this PR: https://github.com/SumoLogic/terraform-provider-sumologic/pull/442. Additionally, as Greg said, new validation does not force sign, it only forces that the old regex matches whole value, not a substring as before.

Reply:
Thanks @MZochowskiSUMO for the explanation.
So as per #442 , we are basically comparing abs values.
regex: `^-?(\d)+[smhd]$` has ? after - which i didn't notice, makes sense that we are not forcing

Reply:
I will mark it then as resolved.